Pod/Classes/WPCarouselAssetsViewController.m

```diff
@@ -0,0 +1,135 @@
+#import "WPCarouselAssetsViewController.h"
+
+@interface WPCarouselAssetsViewController () <UIPageViewControllerDataSource, UIPageViewControllerDelegate, WPAssetViewControllerDelegate>
+@property (nonatomic, strong) NSArray<id<WPMediaAsset>> *assets;
+@property (assign, nonatomic) NSInteger index;
+@property (assign, nonatomic) NSInteger nextIndex;
+@end
+
+@implementation WPCarouselAssetsViewController
+
+- (instancetype)initWithAssets:(NSArray<id<WPMediaAsset>> *)assets
+{
+    NSDictionary *options = @{UIPageViewControllerOptionInterPageSpacingKey : @20};
+    self = [super initWithTransitionStyle:UIPageViewControllerTransitionStyleScroll
+                    navigationOrientation:UIPageViewControllerNavigationOrientationHorizontal
+                                  options:options];
+    if (self) {
+        _assets = assets;
+    }
+
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    [self initialSetup];
+    [self updateTitle];
+}
+
+- (void)setPreviewingAssetAtIndex:(NSInteger)index animated:(BOOL)animated
+{
+    self.index = index;
+    if (self.isViewLoaded) {
+        UIViewController *newViewController = [self viewControllerAtIndex:index];
+        [self setViewController:newViewController animated:animated];
+    }
+}
+
+#pragma mark: - UIPageViewControllerDelegate
```

Comment:
`pragma mark`s don't need a `:` before the `-` – just `#pragma mark - UIPageViewControllerDelegate`. There are a couple in here that need updating.